griptape/drivers/prompt/cohere_prompt_driver.py

```diff
@@ -87,19 +99,145 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
             "temperature": self.temperature,
             "stop_sequences": self.tokenizer.stop_sequences,
             "max_tokens": self.max_tokens,
+            **({"tool_results": tool_results} if tool_results else {}),
+            **(
+                {"tools": self.__to_cohere_tools(prompt_stack.actions), "force_single_step": self.force_single_step}
+                if prompt_stack.actions and self.use_native_tools
+                else {}
+            ),
             **({"preamble": preamble} if preamble else {}),
         }
 
-    def __prompt_stack_content_message_content(self, content: BaseMessageContent) -> dict:
+    def __to_cohere_messages(self, messages: list[Message]) -> list[dict]:
+        cohere_messages = []
+
+        for message in messages:
+            cohere_message: dict = {"role": self.__to_cohere_role(message)}
+
+            if message.has_any_content_type(ActionResultMessageContent):
+                cohere_message["tool_results"] = [
+                    self.__to_cohere_message_content(action_call)
+                    for action_call in message.get_content_type(ActionResultMessageContent)
+                ]
+            else:
+                cohere_message["message"] = message.to_text()
+                if message.has_any_content_type(ActionCallMessageContent):
+                    cohere_message["tool_calls"] = [
+                        self.__to_cohere_message_content(action_call)
+                        for action_call in message.get_content_type(ActionCallMessageContent)
+                    ]
+
+            cohere_messages.append(cohere_message)
+
+        return cohere_messages
+
+    def __to_cohere_message_content(self, content: BaseMessageContent) -> str | dict:
         if isinstance(content, TextMessageContent):
+            return content.artifact.to_text()
+        elif isinstance(content, ActionCallMessageContent):
+            action = content.artifact.value
+
+            return {"name": f"{action.name}_{action.path}", "parameters": action.input}
+        elif isinstance(content, ActionResultMessageContent):
+            artifact = content.artifact
+
+            if isinstance(artifact, ListArtifact):
+                message_content = [{"text": artifact.to_text()} for artifact in artifact.value]
+            else:
+                message_content = [{"text": artifact.to_text()}]
```

Comment:
Do we know this will always be a `TextArtifact` or `ListArtifact`?

Reply:
It's either a `ListArtifact` or some other Artifact that we're going to convert to text as best as we can.